src/VecSim/algorithms/hnsw/hnswlib.h

```diff
@@ -110,8 +110,8 @@ class HierarchicalNSW : public VecsimBaseObject {
     size_t getRandomLevel(double reverse_size);
     vecsim_stl::vector<tableint> *getIncomingEdgesPtr(tableint internal_id, size_t level) const;
     void setIncomingEdgesPtr(tableint internal_id, size_t level, void *set_ptr);
-    linklistsizeint *get_linklist0(tableint internal_id) const;
-    linklistsizeint *get_linklist(tableint internal_id, size_t level) const;
+    inline linklistsizeint *get_linklist0(tableint internal_id) const;
+    inline linklistsizeint *get_linklist(tableint internal_id, size_t level) const;
```

Comment:
there are more functions to inline in the class. consider doing them all or doing them all in another PR. There is an issue already regarding this.